### DISCUSSION TITLE
Use PNP Core SDK behind company proxy

### DISCUSSION THREAD

**Discussion by joaosousa92** [pnp/pnpcore#1039] [General]
Hello,

Sorry if this is not the ideal place, but im trying to connect to O365 tenant behind a company proxy, and it returns me a "No such host is known"

I have already created my custom auth Provider with a HttpClient, but it seems when initializing the PNP Context he uses the default rest cliient in the SharepointRestClient object.

Has anyone encountered this problem?

Thanks in advance.

**Comment by jansenbe** [comment]
@joaosousa92 : the library adds the used `Httpclient` automatically with a custom handler for throttling and ratelimit processing. The .NET HttpClient can be configured to use a proxy server via environment variables (see https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultproxy?view=net-7.0). Can that not be an option for you?

**Reply by joaosousa92** [reply]
Thanks for your reply! I can try that option, but just to make sure, there is no other way to do that through runtime? Like instantiate other HttpClient on the services configuration?

**Reply by jansenbe** [reply]
@joaosousa92 : at the moment there's indeed no option to pass in your HttpClient at configuration time.
